tests/osaka/eip7594_peerdas/helper_functions.py

```diff
@@ -0,0 +1,240 @@
+# our equivalent of ckzg test_recover_cells_and_kzg_proofs
+def eest_delete_cells_then_recover_them(cells: list[bytes], proofs: list[bytes], deletion_indices: list[int]):  # noqa: E501
+    """
+    Simulate the cell recovery process in user-specified scenario.
+
+    Note: Requirement for successful reconstruction is having at least N of the 2N cells.
+
+    Theoretical Usage: You pass a cell list with to 128 elements to this function along with a list of deletion indices.
+    These cells will be deleted and then the ckzg recovery mechanism is used to repair the missing cells.
+    If no assertion is triggered the reconstruction was successful.
+    """  # noqa: E501
+    # sanity checks
+    assert len(cells) == 128, f"You are supposed to pass a full cell list with 128 elements to this function, but got list of length {len(cells)}"  # noqa: E501
+
+    assert len(deletion_indices) < 129, f"You can't delete more than every cell (max len of deletion indices list is 128), but you passed a deletion indices list of length {len(deletion_indices)}"  # noqa: E501
+    for i in deletion_indices:
+        assert 0 <= i <= 127, f"Expected integers in range [0, 127], but got: {i}"
+
+    # delete cells
+    all_cell_indices: list[int] = list(range(128))
+    remaining_indices: list[int] = [i for i in all_cell_indices if i not in deletion_indices]
+    remaining_cells = [c for i, c in enumerate(cells) if i not in deletion_indices]
+
+    # print(f"Cells: {cells}\nDeletion Indices: {deletion_indices}\nRemaining indices: {remaining_indices}\nRemaining cells: {remaining_cells}")  # noqa: E501
+
+    # try to reconstruct cells
+    ts = ckzg.load_trusted_setup("trusted_setup.txt", 0)
+    recovered_cells, recovered_proofs = ckzg.recover_cells_and_kzg_proofs(remaining_indices, remaining_cells, ts) # on success returns two lists of len 128  # noqa: E501
+
+    # determine success/failure
+    assert len(recovered_cells) == len(cells), f"Failed to recover cell list. Original cell list had length {len(cells)} but recovered cell list has length {len(recovered_cells)}"  # noqa: E501
+    assert len(recovered_proofs) == len(proofs), f"Failed to recover proofs list. Original proofs list had length {len(proofs)} but recovered proofs list has length {len(recovered_proofs)}"  # noqa: E501
+
+    for i in range(len(recovered_cells)):
+        assert cells[i] == recovered_cells[i], f"Failed to correctly restore missing cells. At index {i} original cell was {cells[i]} but reconstructed cell does not match: {recovered_cells[i]}"  # noqa: E501
+        assert proofs[i] == recovered_proofs[i], f"Failed to correctly restore missing proofs. At index {i} original proof was {proofs[i]} but reconstructed proof does not match: {recovered_proofs[i]}"  # noqa: E501
+
+    # print("Successful reconstruction")
+
+class PersistentBlobGenerator:
+    # PersistentBlobGenerator("4a") creates blob_4a.json in cwd and contains blob, commitment, cells and proofs
+    encoding: str = "utf-8"
+
```

Comment:
Pydantic can be of excellent help here:
```python
from typing import List
from pydantic import BaseModel
from ethereum_test_base_types import Bytes

class Blob(BaseModel):
    name: str
    blob: Bytes
    commitments: List[Bytes]
    cells: List[Bytes]
    proofs: List[Bytes]
```

And this can be then converted to json using `model_dump` or `model_dump_json`, and the `to_json` and `from_json` methods can be removed.